frontend/src/app/ml-models/ml-model-form/ml-model-form.component.ts

```diff
@@ -301,43 +297,42 @@ export class MlModelFormComponent implements OnInit {
   }
 
   /**
-   * Resets label name and key in the event an update to the available labels causes the currently
-   * selected label to no longer be available. Also resets output settings based on whether the
-   * label is considered a score or revenue.
+   * Handles ensuring the label fields are updated appropriately when form fields that
+   * affect what's allowed to be selected are changed.
    */
   refreshLabel() {
-    const labels = this.labels;
-    const label = this.label;
-
+    const usesFirstPartyData = this.value('usesFirstPartyData');
```

Comment:
Nit. this.value('usesFirstPartyData') is used multiple times so can it be set in an initializer function?
E.g., this.usesFirstPartyData = this.value('usesFirstPartyData');